### PR TITLE
Adds helpers to create, get, and delete ECR repositories

### PR DESCRIPTION
**Pull request by bwhaley**
Adds new functions for interfacing with ECR.

### CODE REVIEW

File: modules/aws/ecr.go
Comment (yorinasub17) [inline modules/aws/ecr.go:12]: NIT: We've been adding a note on the non-E functions to mention that:
```suggestion
// CreateECRRepo creates a new ECR Repository. This will fail the test and stop execution if there is an error.
```

This applies to all the non-E functions in this file.

File: modules/aws/ecr.go
Comment (yorinasub17) [inline modules/aws/ecr.go:45]: NIT: We should return a custom error type here so that you can check for it. E.g., if one wanted to distinguish between error because repo doesn't exist and error because of AWS issues.

File: modules/aws/ecr_test.go
Comment (yorinasub17) [inline modules/aws/ecr_test.go:16]:
```suggestion
	require.NoError(t, err)
```

So we don't do the rest of the steps and panic if there is an error when we dereference the pointer in the next step when there is an error.

This applies to the similar checks done below.

File: modules/aws/ecr_test.go
Comment (yorinasub17) [inline modules/aws/ecr_test.go:17]: NIT: use `aws.String`

File: modules/aws/ecr_test.go
Comment (yorinasub17) [inline modules/aws/ecr_test.go:8]: NIT: If you decide to add a custom error for the non-existent ECR repo, consider adding a unit test that checks for that error type when you call `GetECRRepoE` with a random name.

**Review by yorinasub17 — COMMENTED** [review]
Thanks for porting it over here! Had a bunch of nits I found now that I'm reviewing it with a fresh mind. I know most of this was copy pasted from our internal repo, so feel free to punt on the suggestions.

File: modules/aws/ecr.go
Comment (bwhaley) [reply modules/aws/ecr.go]: Done in 2840220

File: modules/aws/ecr_test.go
Comment (bwhaley) [reply modules/aws/ecr_test.go]: Done in 2840220

File: modules/aws/ecr_test.go
Comment (bwhaley) [reply modules/aws/ecr_test.go]: Done in 2840220

File: modules/aws/ecr.go
Comment (bwhaley) [reply modules/aws/ecr.go]: This error condition should never occur. If no repositories are found, the `DescribeRepositories` call itself will return an error. We can't have more than one response because we're only providing one name to `RepositoryNames`. However, it's never a good idea to index a slice without length checking first, hence this check.. but I'm not sure how to cause this condition, so we also wouldn't be able to check for a custom error. Thus I just stuck with the `fmt.Errorf` (although I did adjust the wording). 

Note that this follows the convention already established throughout `terratest` (e.g. in ecs.go, ssm.go, and elsewhere).

File: modules/aws/ecr_test.go
Comment (bwhaley) [reply modules/aws/ecr_test.go:10]: See above comment on the issue with testing for this condition.

**Comment by bwhaley** [comment]
Thanks for the review. Note that I tried to follow the conventions already established in the package and used ecs.go as a model, but I do understand that the preferences and best practices are evolving. Let me know if you have any further feedback after the most recent updates!

File: modules/aws/ecr.go
Comment (yorinasub17) [reply modules/aws/ecr.go]: Ahhh ok. In that case, the error message should actually mention that this is an impossible case and if a user runs into it, they should file an issue.

File: modules/aws/ecr.go
Comment (yorinasub17) [inline modules/aws/ecr.go:37]: NIT: It's probably worth mentioning that this will error if the ECS repo doesn't exist.

**Review by yorinasub17 — APPROVED** [review]
Thanks for addressing! Had two more nits, but super minor. Feel free to punt to future PR.

File: modules/aws/ecr.go
Comment (bwhaley) [reply modules/aws/ecr.go]: Good thinking, that sounds like the right thing to do. 2e7a21e

File: modules/aws/ecr.go
Comment (bwhaley) [reply modules/aws/ecr.go:38]: 2e7a21e

**Comment by bwhaley** [comment]
Ok, I think I've incorporated all the feedback!

**Review by yorinasub17 — APPROVED** [review]
LGTM!

**Comment by bwhaley** [comment]
Thanks for the reviews and feedback!

**Comment by bwhaley** [comment]
Had to catch up to master before merging - can I get another approval please?

**Review by yorinasub17 — APPROVED** [review]

**Pull request merged**
